garak/detectors/base.py

```diff
@@ -177,3 +178,32 @@ def detect(
                 detector_results.append(1.0 if match else 0.0)
 
         return detector_results
+
+
+class FileDetector(Detector):
+
+    valid_format = "local filename"
+
+    def _test_file(self, filename: str) -> Union[None, str]:
+        raise NotImplementedError
+
+    def detect(self, attempt: garak.attempt.Attempt) -> Iterable[float]:
+        if self.valid_format and (
+            "format" not in attempt.notes
+            or attempt.notes["format"] != self.valid_format
+        ):
+            raise ValueError(
+                f"detectors.fileformats.{self.__class__.__name__} only processes outputs that are '{self.valid_format}'"
+            )
+
+        for local_filename in attempt.outputs:
+            if local_filename is None or local_filename == "":
+                continue
+            if not os.path.isfile(
+                local_filename
+            ):  # skip missing files but also pipes, devices, etc
+                continue
```

Comment:
Thought for future consideration, would there be value in logging a missing filename that the probe reported as finding but is no longer found at detection time?

Reply:
Great idea